### PR TITLE
Add ol.source.ImageArcGISRest

### PR DESCRIPTION
**Pull request by llambanna**
ol.source.ImageArcGISRest is added so a single image can be retrieved
which is required when labelling is applied to the underlying map
service.
it's based on the approach used in ol.source.ImageWMS and ol.source.TileArcGISRest

usage:

```javascript

var layer = new ol.layer.Image({
  source: new ol.source.ImageArcGISRest({
    url: "http://sampleserver1.arcgisonline.com/ArcGIS/rest/services/Specialty/ESRI_StateCityHighway_USA/MapServer",
  })
});


```

### CODE REVIEW

**Comment by ahocevar** [comment]
This branch has conflicts. Closing due to lack of activity. @llambanna, please reopen when you get to working on this again.

**Comment by ahocevar** [comment]
Thanks for your work on this @llambanna. I have reopened this because there seems to be demand for it, and the brandh has no conflicts any more. Any chance you could update this branch to remove the use of `goog.isDef`, `goog.isNull`, `goog.string.endsWith` and `goog.object`?

**Comment by devdattaT** [comment]
@llambanna , @ahocevar  Can I send a PR after conflict resolution and other issues, so that we can get this working?

**Comment by llambanna** [comment]
Yep, I just need to find time to tidy it up, next week with a bit of luck.

**Comment by llambanna** [comment]
@ahocevar as you can see from my commits, i've updated from master, removed the goog usages and updated so it passes the build and tests.

**Comment by ahocevar** [comment]
Thanks a lot @llambanna.

File: externs/olx.js [hunk 1]
```diff
@@ -3094,18 +3094,6 @@ olx.interaction.SelectOptions.prototype.toggleCondition;
...
-
-/**
- * Collection where the interaction will place selected features. Optional. If
- * not set the interaction will create a collection. In any case the collection
- * used by the interaction is returned by
- * {@link ol.interaction.Select#getFeatures}.
- * @type {ol.Collection.<ol.Feature>|undefined}
- * @api
- */
-olx.interaction.SelectOptions.prototype.features;
```
Comment (ahocevar) [inline externs/olx.js]: I think this was accidentally removed?

**Comment by ahocevar** [comment]
Looks very goog @llambanna. Do you think you are able to add back `olx.interaction.SelectOptions.prototype.features` to `externs/olx.js`? This is the only think that keeps me from merging immediately.

**Comment by llambanna** [comment]
@ahocevar  i've added that back in as well as an example

**Comment by ahocevar** [comment]
This is truly awesome @llambanna. Thank you so much for this contribution, and sorry for having closed this without giving it the attention it deserved.

**Pull request merged**